src/meta/src/rpc/ddl_controller.rs

```diff
@@ -789,18 +792,14 @@ impl DdlController {
     pub(crate) async fn validate_cdc_table(
         table: &Table,
         table_fragments: &TableFragments,
-    ) -> MetaResult<()> {
+    ) -> anyhow::Result<()> {
         let stream_scan_fragment = table_fragments
             .fragments
             .values()
             .filter(|f| f.fragment_type_mask & FragmentTypeFlag::StreamScan as u32 != 0)
             .exactly_one()
-            .map_err(|err| {
-                anyhow!(format!(
-                    "expect exactly one stream scan fragment, got {}",
-                    err
-                ))
-            })?;
+            .map_err(|e| anyhow::anyhow!("{}", e.to_report_string()))
```

Comment:
Seems not necessary?

Reply:
🤔

Reply:
Oh, I recalled why. `exactly_one` returns an error containing the original iterator. So it will returns lifetime too short error without `map_err`. 🤣

Reply:
`.ok().context(..)` seems good enough. 😄